scripts/linking.py

```diff
@@ -597,13 +620,18 @@ def eval_candidate_generation_and_linking(examples: List[data_util.MedMentionExa
                 batch_candidate_neighbor_ids = candidate_generator.generate_candidates(mention_texts, k)
 
                 filtered_batch_candidate_neighbor_ids = []
-                for candidate_neighbor_ids in batch_candidate_neighbor_ids:
+                for candidate_neighbor_ids, mention_text, predicted_entity, predicted_mention_type \
+                    in zip(batch_candidate_neighbor_ids, mention_texts, predicted_entities, predicted_mention_types):
                     # Keep only canonical entities for which at least one mention has a score less than the threshold.
                     filtered_ids = {k: v for k, v in candidate_neighbor_ids.items() if any([z[1] <= threshold for z in v])}
                     filtered_batch_candidate_neighbor_ids.append(filtered_ids)
                     num_candidates.append(len(candidate_neighbor_ids))
                     num_filtered_candidates.append(len(filtered_ids))
                     doc_candidates.update(filtered_ids)
+                    if len(filtered_ids) != 0:
+                        sorted_candidate_ids = linker.link(filtered_ids, mention_text, predicted_mention_type)
```

Comment:
why is this being called here ?

Reply:
the main loop below runs over gold entities. to get the full set of predicted entities for a document, we need to loop over all predicted entities and link them. There is certainly a refactor that would allow for only calling link once for each predicted entity, but I didn't do that. And it doesn't seem to slow it down all that much.

Reply:
@danielkingai2 , I see. Can add a comment clarifying that part ?